### PR TITLE
max sum array

### LINKED ISSUE #732: Largest Sum Contiguous Sub-Array in Python3
This is an open issue for the Largest Sum Contiguous Sub-Array algorithm implementation in Python3. If you wish to contribute, just make some noise and we will assign you.

### PR DESCRIPTION
**Pull request by LON3WO7F**
**Fixes #732  **

By submitting this pull request I confirm I've read and complied with the below declarations.

-   [x] I have read the [Contribution guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md) 
and I am confident that my PR reflects them.

-   [x] I have followed the [coding guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#cs) 
for this project.

-   [x] My code follows the [skeleton code structure](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#sample).

-   [x] This pull request has a descriptive title. 
For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.

-   [x] This pull request shall only be merged once the build passes.

-   [x] I have mentioned the issue number correctly (with hyper-link) in this pull request description.

### CODE REVIEW

**Comment by Danushka96** [comment]
Hi @LON3WO7F 
You need to get assigned for an issue before making a PR,

**Comment by LON3WO7F** [comment]
Hey

> Hi @LON3WO7F
> You need to get assigned for an issue before making a PR,

Hey,  @Danushka96 You already assigned this to me.
Screenshot from 2020-10-25 16-15-01

**Comment by Danushka96** [comment]
Hi @LON3WO7F 
Sorry my bad, Please link the issue to the PR,

**Comment by Danushka96** [comment]
The file should have an extension, please read the contribution guidelines first!

File: algorithms/max_sum_array.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class max_sum_array(object):
+   def maxSubArray(self, nums):
+
+      num_array = [0 for i in range(len(nums))]
+      num_array[0] = nums[0]
+      for i in range(1,len(nums)):
+         num_array[i] = max(num_array[i-1]+nums[i],nums[i])
+
+      return max(num_array)
+nums = [-13, -3, -25, -20,2,1,-3,7,-2,2,1,-5, -3, -16]
```
Comment (RavinduSachintha) [inline algorithms/max_sum_array.py:10]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6396358)

File: algorithms/max_sum_array.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class max_sum_array(object):
+   def maxSubArray(self, nums):
```
Comment (RavinduSachintha) [inline algorithms/max_sum_array.py:2]: Codacy found an issue: [Method could be a function](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6396358)

File: algorithms/maximum_sum_array.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+class max_sum_array(object):
+   def maxSubArray(self, nums):
```
Comment (RavinduSachintha) [inline algorithms/maximum_sum_array.py:2]: Codacy found an issue: [Method could be a function](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6396358)

File: algorithms/maximum_sum_array.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+      num_array = [0 for i in range(len(nums))]
+      num_array[0] = nums[0]
+      for i in range(1,len(nums)):
+         num_array[i] = max(num_array[i-1]+nums[i],nums[i])
+
+      return max(num_array)
+nums = [-13, -3, -25, -20,2,1,-3,7,-2,2,1,-5, -3, -16]
```
Comment (RavinduSachintha) [inline algorithms/maximum_sum_array.py:10]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6396358)

File: algorithms/maximum_sum_array.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
+def maxSubArray(nums):
+       num_array = [0 for i in range(len(nums))]
+       num_array[0] = nums[0]
+       for i in range(1,len(nums)):
+              num_array[i] = max(num_array[i-1]+nums[i],nums[i])
+
+       return max(num_array)
+
+
+nums = [-13, -3, -25, -20,2,1,-3,7,-2,2,1,-5, -3, -16]
```
Comment (RavinduSachintha) [inline algorithms/maximum_sum_array.py:10]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6396358)

File: algorithms/ar-lscsa/python3/max_sum_array.py [hunk 6]
```diff
@@ -0,0 +1,10 @@
+def maxSubArray(nums):
+       num_array = [0 for i in range(len(nums))]
+       num_array[0] = nums[0]
+       for i in range(1,len(nums)):
+              num_array[i] = max(num_array[i-1]+nums[i],nums[i])
+       return max(num_array)
+
+
+nums = [-13, -3, -25, -20,2,1,-3,7,-2,2,1,-5, -3, -16]
```
Comment (RavinduSachintha) [inline algorithms/ar-lscsa/python3/max_sum_array.py:9]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6396358)

File: algorithms/maximum_sum_array.py [hunk 7]
```diff
@@ -0,0 +1,10 @@
+def maxSubArray(nums):
+       num_array = [0 for i in range(len(nums))]
+       num_array[0] = nums[0]
+       for i in range(1,len(nums)):
+              num_array[i] = max(num_array[i-1]+nums[i],nums[i])
+       return max(num_array)
+
+
+nums = [-13, -3, -25, -20,2,1,-3,7,-2,2,1,-5, -3, -16]
```
Comment (RavinduSachintha) [inline algorithms/maximum_sum_array.py:9]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6396358)

**Comment by LON3WO7F** [comment]
@RavinduSachintha  I don't know where is the error.

File: algorithms/maximum_sum_array.py (hunk 7 above)
Comment (RavinduSachintha) [inline algorithms/maximum_sum_array.py:9]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6396358)

File: algorithms/ar-lscsa/python3/max_sum_array.py [hunk 8]
```diff
@@ -0,0 +1,8 @@
+def maxSubArray(nums):
+       num_array = [0 for i in range(len(nums))]
+       num_array[0] = nums[0]
+       for i in range(1,len(nums)):
+              num_array[i] = max(num_array[i-1]+nums[i],nums[i])
+       return max(num_array)
+nums = [-13, -3, -25, -20,2,1,-3,7,-2,2,1,-5, -3, -16]
```
Comment (RavinduSachintha) [inline algorithms/ar-lscsa/python3/max_sum_array.py:7]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6396358)

File: algorithms/maximum_sum_array.py (hunk 7 above)
Comment (Danushka96) [reply algorithms/maximum_sum_array.py:9]: @LON3WO7F
line number 9

File: algorithms/ar-lscsa/python3/max_sum_array.py (hunk 8 above)
Comment (Danushka96) [reply algorithms/ar-lscsa/python3/max_sum_array.py:7]: @LON3WO7F
line number 7

**Comment by Danushka96** [comment]
Hi @LON3WO7F
The location to put your files should be 
[https://github.com/FOSS-UCSC/FOSSALGO/tree/master/algorithms/ar-lscsa/python3](https://github.com/FOSS-UCSC/FOSSALGO/tree/master/algorithms/ar-lscsa/python3)

**Comment by LON3WO7F** [comment]
> Hi @LON3WO7F
> The location to put your files should be
> [/algorithms/ar-lscsa/python3](https://github.com/FOSS-UCSC/FOSSALGO/tree/master/algorithms/ar-lscsa/python3)

Screenshot from 2020-10-26 09-04-43

Hey @Danushka96 It is already in the right path.

File: algorithms/ar-lscsa/python3/max_sum_array.py [hunk 9]
```diff
@@ -0,0 +1,8 @@
+def maxSubArray(nums):
+       num_array = [0 for i in range(len(nums))]
+       num_array[0] = nums[0]
+       for i in range(1,len(nums)):
+              num_array[i] = max(num_array[i-1]+nums[i],nums[i])
+       return max(num_array)
+nums = [-13,-3,-25,-20,2,1,-3,7,-2,2,1,-5,-3,-16]
```
Comment (RavinduSachintha) [inline algorithms/ar-lscsa/python3/max_sum_array.py:7]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6396358)

File: algorithms/maximum_sum_array.py (hunk 7 above)
Comment (RavinduSachintha) [inline algorithms/maximum_sum_array.py:9]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6396358)

**Comment by Danushka96** [comment]
@LON3WO7F 
There are two files, one is in the right path and another one is wrong

File: algorithms/maximum_sum_array.py [hunk 10]
```diff
@@ -0,0 +1,10 @@
+def maxSubArray(nums):
```
Comment (Danushka96) [inline algorithms/maximum_sum_array.py:1]: Wrong path

**Review by Danushka96 — CHANGES_REQUESTED** [review]

File: algorithms/ar-lscsa/python3/max_sum_array.py (hunk 9 above)
Comment (RavinduSachintha) [inline algorithms/ar-lscsa/python3/max_sum_array.py:7]: Codacy found an issue: [Trailing whitespace](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6396358)

Comment (Danushka96) [reply algorithms/ar-lscsa/python3/max_sum_array.py:7]: @LON3WO7F 
There a space at the end of line 7, remove it, it will fix this issue

**Comment by LON3WO7F** [comment]
Screenshot from 2020-10-26 09-43-27
Hey @Danushka96 it is showing me **_requested changes._**
How i'm gonna apply changes and i can not see what type of changes you want

**Review by Danushka96 — APPROVED** [review]

**Comment by Danushka96** [comment]
Hi @LON3WO7F 
Can you update the main Readme.md file with the algorithm you implemented?

**Comment by Danushka96** [comment]
@all-contributors please add @LON3WO7F for code and doc

**Comment by Danushka96** [comment]
Hi @LON3WO7F 
Thanks for your contribution!

**Pull request merged**